File: backend/app/api/routes/businesses.py
from fastapi import APIRouter, Query, Path
from typing import Optional
from app.schemas.business import BusinessCreate, BusinessUpdate, BusinessResponse, BusinessListResponse
from app.services.business_service import business_service

router = APIRouter()

@router.post("", response_model=dict, status_code=201)
async def create_business(business_in: BusinessCreate):
    created = await business_service.create_business(business_in)
    return {
        "success": True,
        "message": "Business created successfully",
        "data": created
    }

@router.get("", response_model=BusinessListResponse)
async def get_businesses(
    page: int = Query(1, ge=1),
    limit: int = Query(20, ge=1, le=100),
    search: Optional[str] = Query(None)
):
    result = await business_service.list_businesses(page, limit, search)
    return {
        "success": True,
        "data": result["data"],
        "pagination": result["pagination"]
    }

@router.get("/by-slug/{slug}", response_model=dict)
async def get_business_by_slug(slug: str = Path(...)):
    business = await business_service.get_business_by_slug(slug)
    return {
        "success": True,
        "message": "Business retrieved successfully",
        "data": business
    }

@router.get("/{business_id}", response_model=dict)
async def get_business(business_id: str = Path(...)):
    business = await business_service.get_business(business_id)
    return {
        "success": True,
        "message": "Business retrieved successfully",
        "data": business
    }

@router.put("/{business_id}", response_model=dict)
async def update_business(
    business_in: BusinessUpdate,
    business_id: str = Path(...)
):
    updated = await business_service.update_business(business_id, business_in)
    return {
        "success": True,
        "message": "Business updated successfully",
        "data": updated
    }

@router.delete("/{business_id}", response_model=dict)
async def delete_business(business_id: str = Path(...)):
    await business_service.delete_business(business_id)
    return {
        "success": True,
        "message": "Business deleted successfully"
    }

File: frontend/lib/api/businesses.ts
import { apiClient } from './client';
import { 
  BusinessCreateInput, 
  BusinessUpdateInput, 
  BusinessListResponse, 
  BusinessResponse 
} from '@/types/business';

export const businessesApi = {
  getBusinesses: async (page = 1, limit = 20, search = '', signal?: AbortSignal): Promise<BusinessListResponse> => {
    const params = new URLSearchParams({
      page: page.toString(),
      limit: limit.toString(),
    });
    if (search) params.append('search', search);
    
    const response = await apiClient.get<BusinessListResponse>(`/businesses?${params.toString()}`, { signal });
    return response.data;
  },

  getBusiness: async (id: string): Promise<BusinessResponse> => {
    const response = await apiClient.get<BusinessResponse>(`/businesses/${id}`);
    return response.data;
  },

  getBusinessBySlug: async (slug: string): Promise<BusinessResponse> => {
    const response = await apiClient.get<BusinessResponse>(`/businesses/by-slug/${slug}`);
    return response.data;
  },

  createBusiness: async (data: BusinessCreateInput): Promise<BusinessResponse> => {
    const response = await apiClient.post<BusinessResponse>('/businesses', data);
    return response.data;
  },

  updateBusiness: async (id: string, data: BusinessUpdateInput): Promise<BusinessResponse> => {
    const response = await apiClient.put<BusinessResponse>(`/businesses/${id}`, data);
    return response.data;
  },

  deleteBusiness: async (id: string): Promise<{ success: boolean; message: string }> => {
    const response = await apiClient.delete(`/businesses/${id}`);
    return response.data;
  }
};
